NBISweden/LocalEGA
# Improve image base and CI build

Separate the OS (packages + python install) from actual source code in order to streamline build based on an idea by @silverdaz, an effort to improve: https://github.com/NBISweden/LocalEGA/pull/348
Part of this PR:
* bumped CentOS to `7.5.1804`
* Split images into `ega-os`, `ega-base`, `ega-openssh` and `ega-inbox` 
   * considering that the packages and the OpenSSH SFTP server are not so frequently built, we decided to split them and not include them in the build every time. The `ega-os` and  `ega-openssh` should be built only if the `Dockerfile` has changed, we use `extras/travis.sh` for this.
   * `ega-base` the LocalEGA app, probably the most frequently changed.
   * `ega-inbox` makes use of of [multi stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) and uses `ega-base` and `ega-openssh` images. Should be built as the `ega-base` changes.
* documentation update

> Do we need to make changes to deployment options?

No. Image names and configuration stayed the same.

> How to test this?

As it stands check the `extras/travis.sh` and after merge we should do two additional PRs one with a `images/os/Dockerfile` change and one without.

> Would be nice of the rabbitmq version got bumped.

Yes indeed, but there was a bug https://github.com/NBISweden/LocalEGA/issues/225 and we fixed the version. In the meantime rabbitmq 3.7 changed their configuration formatting: http://www.rabbitmq.com/blog/2018/02/22/new-configuration-format-in-rabbitmq-3-7/ . If enough interest and depending on CEGA version we might change it. Mostly because of security updates.

> One integration test fails.

We should increase travis `sleep`, run again that job or be smart about it and modify the integration tests and make it wait for services to be up.
